qemu/qmp.go

```diff
@@ -178,6 +178,36 @@ type MemoryDevices struct {
 	Type string            `json:"type"`
 }
 
+// CPUInstanceProperties represent cpu properties
+type CPUInstanceProperties struct {
+	NodeID   uint64 `json:"node-id"`
+	SocketID uint64 `json:"socket-id"`
+	CoreID   uint64 `json:"core-id"`
+	ThreadID uint64 `json:"thread-id"`
+}
+
+// CPUInfo represents information about each virtual CPU
+type CPUInfo struct {
+	CPU      int64                 `json:"CPU"`
+	Current  bool                  `json:"current"`
+	Halted   bool                  `json:"halted"`
+	QomPath  string                `json:"qom_path"`
+	Arch     string                `json:"arch"`
+	Pc       int64                 `json:"pc"`
```

Comment:
Do we want to be explicit about the sizes of the integer types.  In general, I don't think we've done this elsewhere.  For example, as far as I can tell QMP defines ThreadID as an "int".  Should we not then use int here?

Reply:
Good point. I can't be sure, but `int` should be enough I guess